issue-2937-button-text-select-error

Marko64 opened the pull request:
# Description

Problem: _Text direction_ is causing an error. It needs a default value.
I'm not sure if this is the correct way to fix it. If not, please suggest where else can I define the defaults.

Fixes #2937

# How Has This Been Tested?

Insert the text or the button block and select it. Check the console for errors. 

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Inset the text block. Select it. Check the console for errors.
-   [ ] Select the Typography panel from the sidebar and test all the settings that use SelectControl. 
-   [ ] Repeat the same for the button block.


**_ Pre-Code Testing _**

-   [x] Inset the text block. Select it. Check the console for errors.
-   [x] Select the Typography panel from the sidebar and test all the settings that use SelectControl. Make sure they work and don't throw any errors in the console.
-   [x] Repeat the same for the button block.

# Checklist

-   [x] I have performed a self-review of my code
-   [x] My changes generate no new warnings/errors

Marko64 committed:
Added defaults

Fixes #2937

Problem: Text direction is causing the error. It needs a default value.
I'm not sure if this is the correct way to fix it. If not, please suggest where else can I define the default.

Naaaaiix reviewed (commented):
This implementation 'fixes' partly the issue, the problem is that can generate a new one. Adding a default attribute we create the CSS, which can go against the creators interests when is rtl website. A better fix would be, on:
https://github.com/yeahcan/maxi-blocks/blob/e489d3e3d300646b2a55f080a3108946d18f9955/src/components/typography-control/index.js#L876

replace for:
```
			<SelectControl
				label={__('Text direction', 'maxi-blocks')}
				className='maxi-typography-control__direction'
				value={getValue(
					`${prefix}text-direction`,
					breakpoint,
					false,
					true
				)}
[...]
````

Marko64 committed:
SelectControl update

Marko64 committed:
Duplicate attr and text-transform

Marko64 committed:
Merge branch 'master' of https://github.com/yeahcan/maxi-blocks into issue-2937-button-selector-error

Marko64 committed:
Added default SC values

Ok, so I think I figured out what's happening. When I added "true" for the avoidSC prop, the value wasn't null anymore, because it was getting it from the SelectControl. Ofc, SC values weren't working anymore. So the issue was that we didn't have default values for our default SC for text orientation and direction so it didn't have any values to show in the first place.

Naaaaiix reviewed (commented):
Tests need an update 😸

Marko64 committed:
Update paragraph.spec.js.snap

Marko64 committed:
Updated tests and snapshots

Naaaaiix committed:
Fixes tests
